### PR TITLE
Fix rn config file load codegen

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## Summary

Resolves https://github.com/facebook/react-native/issues/34894

## Changelog

[General][Fixed] - Load react-native.config.js from correct path during codegen

## Test Plan

Command line output after fix with extra console logs to show the correctness ( hopefully :) ) of the fix.

```bash
 ~/repos/sentry-react-native   krystofwoldrich/multipleRNsamples ± node node_modules/react-native/scripts/generate-artifacts.js --path sample-new-architecture --outputPath codegen
[Codegen] Processing react-native core libraries
[Codegen] Found react-native


[Codegen] >>>>> Searching for codegen-enabled libraries in /Users/krystofwoldrich/repos/sentry-react-native/node_modules


[Codegen] >>>>> Searching for codegen-enabled libraries in react-native.config.js
cwd /Users/krystofwoldrich/repos/sentry-react-native
rnConfigFilePath sample-new-architecture/react-native.config.js
rnConfigFilePath - resolved /Users/krystofwoldrich/repos/sentry-react-native/sample-new-architecture/react-native.config.js
[Codegen] Found @sentry/react-native


[Codegen] >>>>> Searching for codegen-enabled libraries in the app


[Codegen] >>>>> Processing FBReactNativeSpec
[Codegen] Generated schema: /var/folders/tl/jddrmdy97gj0cljrcwb_qkzc0000gn/T/FBReactNativeSpecrHeknw/schema.json
[Codegen] Generated artifacts: codegen/build/generated/ios


[Codegen] >>>>> Processing rncore
[Codegen] Generated schema: /var/folders/tl/jddrmdy97gj0cljrcwb_qkzc0000gn/T/rncorehCjmlJ/schema.json
[Codegen] Generated artifacts: codegen/build/generated/ios


[Codegen] >>>>> Processing RNSentrySpec
[Codegen] Generated schema: /var/folders/tl/jddrmdy97gj0cljrcwb_qkzc0000gn/T/RNSentrySpectMmDpf/schema.json
[Codegen] Generated artifacts: codegen/build/generated/ios


>>>>> Creating component provider
Generated schema list: /var/folders/tl/jddrmdy97gj0cljrcwb_qkzc0000gn/T/rn-tmp-schema-list.json
Generated provider in: codegen/build/generated/ios


[Codegen] Done.
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## Unreleased
```
Comment (cipolleschi) [inline CHANGELOG.md:3]: Thank you for this, but there is no need to update the changelog! ;)

**Review by cipolleschi — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (krystofwoldrich) [reply CHANGELOG.md]: Okay :D I'll remove it. Thanks

**Review by cipolleschi — APPROVED** [review]

**Comment by krystofwoldrich** [comment]
Thank you for the review. Can you merge it?

**Pull request closed without merging**